android/sdl_android/src/main/java/com/smartdevicelink/SdlConnection/SdlSession.java

```diff
@@ -1,838 +1,338 @@
+            return new SdlProtocol(this, (TCPTransportConfig) transportConfig);
+        }
+
+        return null;
+    }
+
+    boolean isAudioRequirementMet() {
+        if (mediaStreamingStatus == null && contextWeakReference != null && contextWeakReference.get() != null) {
+            mediaStreamingStatus = new MediaStreamingStatus(contextWeakReference.get(), new MediaStreamingStatus.Callback() {
+                @Override
+                public void onAudioNoLongerAvailable() {
+                    close();
+                    shutdown("Audio output no longer available");
+                }
+            });
+        }
+
+        // If requiresAudioSupport is false, or a supported audio output device is available
+        return !requiresAudioSupport || mediaStreamingStatus.isAudioOutputAvailable();
+
+    }
+
+
+    @SuppressWarnings("RedundantThrows")
+    @Override
+    public void startSession() throws SdlException {
+        if (!isAudioRequirementMet()) {
+            shutdown("Audio output not available");
+            return;
+        }
+
+        sdlProtocol.start();
+    }
+
+    @Override
+    public TransportType getCurrentTransportType() {
+        return TransportType.MULTIPLEX;
+    }
+
+    @Override
+    public void shutdown(String info) {
+        DebugTool.logInfo(TAG, "Shutdown - " + info);
+        if (mediaStreamingStatus != null) {
+            mediaStreamingStatus.clear();
+        }
+        super.shutdown(info);
+
+    }
+
+    /**
+     * Get the current protocol version used by this session
+     *
+     * @return Version that represents the Protocol version being used
+     */
+    @Override
+    public Version getProtocolVersion() {
+        if (sdlProtocol != null) {
+            return sdlProtocol.getProtocolVersion();
+        }
+        return new Version(1, 0, 0);
+    }
+
+
+    /* ***********************************************************************************************************************************************************************
+     * *****************************************************************  IProtocol Listener  ********************************************************************************
+     *************************************************************************************************************************************************************************/
+
+    @Override
+    public void onProtocolMessageBytesToSend(SdlPacket packet) {
+        sdlProtocol.sendPacket(packet);
+    }
+
+    @Override
+    public void onProtocolSessionStarted(SessionType sessionType,
+                                         byte sessionID, byte version, String correlationID, int hashID, boolean isEncrypted) {
+
+        DebugTool.logInfo(TAG, "Protocol session started");
+
+        this.sessionId = sessionID;
+        if (sessionType.eq(SessionType.RPC)) {
+            sessionHashId = hashID;
+        }
+        if (isEncrypted)
```

Comment:
I missed this on the first pass, Since we changed it in JavaSE, we should add brackets here as well